src/main/java/org/traccar/helper/CoordinateUtil.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright 2024 - 2024 容均致 (harryrong@rushanio.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.helper;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Coordinate system conversion related toolkits, mainstream coordinate systems include:
+ * WGS84 coordinate system: colloquially Earth coordinate system, used by Maps outside China.
+ * GCJ02 coordinate system: colloquially Mars coordinate system, used by Gaode, Tencent, Ali and so on.
+ * <p>
+ * Reference：<a href="https://en.wikipedia.org/wiki/Restrictions_on_geographic_data_in_China">...</a>
+ * Reference: <a href="https://github.com/dromara/hutool">...</a>
+ */
+public final class CoordinateUtil {
+
+    private CoordinateUtil() {
+
+    }
+
+    /**
+     * Coordinate conversion parameters：π（Pi)
+     */
+    public static final double PI = 3.1415926535897932384626433832795D;
```

Comment:
Can we use `Math.PI` instead of this?

Reply:
Yes, I have resubmitted it. Although the PI accuracy in CoordinateUtil.java is higher than Math.PI, it has been tested, but Math.PI is accurate enough to be used in the current business.